composables/useCollectionAttributes.ts

```diff
@@ -0,0 +1,64 @@
+import type { NFT, Attribute } from '@/types'
+
+export const useCollectionAttributes = (collectionId: ComputedRef<string | undefined>) => {
+  const { data: nftAttributesData, refetch } = useGraphql({
+    queryName: 'nftAttributesListByCollection',
+    variables: {
+      id: collectionId.value,
+    },
+    disabled: computed(() => !collectionId.value),
+  })
+
+  watch(collectionId, () => {
+    refetch({
+      id: collectionId.value,
+    })
+  })
+
+  const nftsList = computed<NFT[]>(() => {
+    return (nftAttributesData.value as unknown as { nfts: NFT[] })?.nfts || []
+  })
+
+  const attributesList = computed<Attribute[]>(() => {
+    return nftsList.value.reduce((acc, nft) => {
+      if (nft.meta?.attributes?.length) {
+        acc.push(...nft.meta.attributes)
+      }
+      return acc
+    }, [] as Attribute[])
+  })
+
+  const attributesRarityMaps = computed(() => {
+    const attributeCounts: Record<string, Record<string, number>> = {}
+
+    attributesList.value.forEach((attr) => {
+      if (!attr.trait) return
+      if (!attributeCounts[attr.trait]) {
+        attributeCounts[attr.trait] = {}
+      }
+
+      attributeCounts[attr.trait][attr.value]
+        = (attributeCounts[attr.trait][attr.value] || 0) + 1
+    })
+    const rarityMaps: Record<string, Record<string, number>> = {}
+    const totalNfts = nftsList.value?.length || 0
```

Comment:
should always return an array 

```suggestion
    const totalNfts = nftsList.value.length
```

Reply:
updated